storybookjs/storybook
# CLI: Workaround for react native `sb init`

Issue:  #12334 

Also various issues from http://github.com/storybookjs/react-native are probably resolved by this

## What I did

I changed the react native generator to work specifically for react native. There are currently some special cases for react native that aren't worth implementing on the base generator because they could be solved with a v6 of react native.

I also made some changes to simplify the examples.

## How to test my changes

You can try out the changes by generating a new react native project and then running the cli index.js with node from inside the react native project. Here is a brief explanation on how you can do that:

from this repo build the cli
```
cd lib/cli
yarn prepare
```

Install the expo cli (if you already have expo cli installed you can skip this step)
```
npm i -g expo-cli
```

Generate a react native project with expo. 
```
expo init --template blank myNewProject
```

It would be ideal to do this near the storybook repo so the file path is easier to figure out

Within the terminal navigate to the react native project `cd myNewProject`

Run the local build of the cli
```
node PATH_T0_STORYBOOK_REPO/lib/cli/bin/index.js init --type react_native
```
choose yes when prompted to install the storybook server

the package.json should have the following dependencies added
```
"@storybook/addon-actions": "^5.3",
"@storybook/addon-knobs": "^5.3",
"@storybook/addon-links": "^5.3",
"@storybook/addon-ondevice-actions": "^5.3.21",
"@storybook/addon-ondevice-knobs": "^5.3.21",
"@storybook/react-native": "^5.3.21",
"@storybook/react-native-server": "^5.3.21",
"babel-loader": "^8.1.0",
```

the storybook command should also be added to the scripts in package json

```
"storybook": "start-storybook -p 7007",
```

To run the project delete the contents of app.js and add the following `export {default} from './storybook';`

Then you can check the ondevice ui with `yarn web` or `yarn ios` or `yarn android`. 
If you don't have a simulator setup I recommend using `yarn web` since it should work regardless.

Check the server with `yarn storybook`
